--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,7 +222,7 @@
 	list(APPEND CORE_VFS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/util/vfs/vfs-fd.c ${CMAKE_CURRENT_SOURCE_DIR}/src/platform/windows/vfs-w32.c)
 	file(GLOB OS_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/platform/windows/*.c)
 	source_group("Windows-specific code" FILES ${OS_SRC})
-elseif(UNIX)
+elseif(UNIX OR CMAKE_SYSTEM_NAME STREQUAL "Android")
 	set(USE_PTHREADS ON)
 
 	if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
